QEP-label-path.rst

```diff
@@ -0,0 +1,71 @@
+.. _qep#[.#]:
+
+========================================================================
+QGIS Enhancement ??: Labeling path
+========================================================================
+
+:Date: October 2015
+:Author: Hugo Mercier
+:Contact: hugo dot mercier at oslandia dot com
+:Last Edited: 
+:Status:  
+:Version:
+
+Summary
+-------
+
+A new labeling option is proposed for addition here. It allows to use a user-defined linear path for the drawing of a label.
+
+Using a simplified path for the curved placement of labels is one of the possible use cases of such a feature.
+
+A user-defined path can be seen as an additional geometry on a layer, where the underlying model supports only one "geometry column".
+
+Proposed solution
+-----------------
+
+We propose the add a way for the user to use a different geometry object as support for the "curved" placement of a label.
+
+The current (2.11) version of QGIS allows to display labels along the linestring features in a "curved" manner.
+
+.. image:: curved_labeling.png
+
+However for linestrings defined with lots of control points (such as waterways) or with high curvatures, displaying correct
+labels can be either unesthetic or very slow.
+
+This is where using a path slightly different from the feature's geometry could be useful. For instance using a simplified version
+of the linestring could lead to a label faster to display or with smaller curvatures.
+
+We then propose to allow the "curved" placement to use an arbitrary geometry, through a data-defined "label path" property. The data-defined
+property could point to a user-defined column or to an expression.
+
+We also propose to add a "simplify()" function in core expression functions for that purpose.
```

Comment:
I've played with this a bit in the past and found that the default Douglas–Peucker simplification method doesn't help at all for curved labeling, as the sharp corners in the linear geometry are preserved. From my tests a better approach is:
- Use Visvalingam’s algorithm to simplify, as that removes sharp corners whilst preserving the overall shape of a line
- Combine smoothing with simplification (see http://qgis.org/api/classQgsGeometry.html#ab2f6d720b51f96686f3152a6dabfe201 ). Note that smoothing increases the number of vertices in a geometry, so it needs to be done after simplifying the geometry.
- Even better would be a "bend simplify" based on the algorithm described in http://web.cs.sunyit.edu/~poissad/projects/Curve/pdf/1998.pdf . It's a more effective method of maintaining the general shape of lines will removing small fluctuations.